x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -293,7 +295,10 @@ export type BulkEditOperation =
       value?: undefined;
     };
 
-type RuleParamsModifier<Params extends RuleTypeParams> = (params: Params) => Promise<Params>;
+type RuleParamsModifier<Params extends RuleTypeParams> = (params: Params) => Promise<{
+  modifiedParams: Params;
+  isParamsUpdateSkipped: boolean;
+}>;
```

Comment:
The `RuleParamsModifier` function used here in the rules_client is not exactly the same as used in `security_solution/server/lib/detection_engine/rule_management/logic/bulk_actions/rule_params_modifier.ts` since this one just receives the `params`, while the one on our side receives the params **and the operations**. Check how that is used here.

However, I extracted out the Result of the `RuleParamsModifier` like so:
```export interface RuleParamsModifierResult<Params> {
  modifiedParams: Params;
  isParamsUpdateSkipped: boolean;
}

type RuleParamsModifier<Params extends RuleTypeParams> = (
  params: Params
) => Promise<RuleParamsModifierResult<Params>>;
```
and could use it in `rule_params_modifier.ts`, like so:
```
export const ruleParamsModifier = (
  existingRuleParams: RuleAlertType['params'],
  actions: BulkActionEditForRuleParams[]
): RuleParamsModifierResult<RuleAlertType['params']> => {
```

Reply:
@jpdjere That LGTM except let's please remove this `as RuleParamsModifier<RuleAlertType['params']>` - `as` is evil and should be avoided by all means :)